### PR TITLE
add CNB gsoc 2023 projects

### PR DESCRIPTION
**Pull request by joe-kimmel-vmw**
Signed-off-by: Joe Kimmel <86852107+joe-kimmel-vmw@users.noreply.github.com>

### CODE REVIEW

**Review by nate-double-u — DISMISSED** [review]
Thanks for this @joe-kimmel-vmw!

Could you add email addresses and github user ids to the lists of mentors?

**Comment by nate-double-u** [comment]
> Could you add email addresses and github user ids to the lists of mentors?

Sorry, didn't realize this was a draft PR before I added my comments

File: summerofcode/2023.md [hunk 1]
```diff
@@ -26,12 +26,42 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
 
 ## Ideas
 
+- [Cloud Native Buildpacks](#cloud-native-build-packs)
```
Comment (nate-double-u) [inline summerofcode/2023.md:29]: let's remove the ToC stuff for now. We can add it back at the end of the process.

File: summerofcode/2023.md [hunk 2]
```diff
@@ -26,12 +26,42 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Description: Widely anticipated since [the 1450s](https://en.wikipedia.org/wiki/Missa_L%27homme_arm%C3%A9), the rise of ARM processors has created new binary targets for pre-compiled executables. Creating these executables can be a valuable educational lesson about where assumptions get baked in, and what's really the difference between x86 and ARM anyhow?
+- Expected outcome: Improved ARM support
+- Recommended Skills: Golang, software development literacy.
+- Mentor(s): Aidan Delaney (@AidanDelaney); Jerico Pena (@jpena-r7)
+- Expected project size: 350 Hours
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/buildpacks/pack/issues/1459
+
+#### Something about Dockerfiles
+- Description: Build out samples and workflows showing how to use Dockerfiles in
```
Comment (natalieparellano) [inline summerofcode/2023.md:58]:
```suggestion
- Description: Recent [changes](https://github.com/buildpacks/rfcs/blob/main/text/0105-dockerfiles.md) to the Cloud Native Buildpacks specification make buildpacks builds more flexible by allowing base image customization with Dockerfiles. [Initial support](https://github.com/buildpacks/rfcs/issues/224) for this feature has been added to the CNB lifecycle and the `pack` CLI, but more work is needed to deliver a seamless end-user experience. Working on this feature will give contributors an in-depth understanding of buildpacks internals and an appreciation for the benefits buildpacks offer - and how buildpacks and Dockerfiles can work together.
```

File: summerofcode/2023.md [hunk 3]
```diff
@@ -26,12 +26,42 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+  harmony with a cloud native buildpacks platform.
```
Comment (natalieparellano) [inline summerofcode/2023.md:59]:
```suggestion
- Description: Recent [changes](https://github.com/buildpacks/rfcs/blob/main/text/0105-dockerfiles.md) to the Cloud Native Buildpacks specification make buildpacks builds more flexible by allowing base image customization with Dockerfiles. [Initial support](https://github.com/buildpacks/rfcs/issues/224) for this feature has been added to the CNB lifecycle and the `pack` CLI, but more work is needed to deliver a seamless end-user experience. Working on this feature will give contributors an in-depth understanding of buildpacks internals and an appreciation for the benefits buildpacks offer - and how buildpacks and Dockerfiles can work together.
```

File: summerofcode/2023.md [hunk 4]
```diff
@@ -26,12 +26,42 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Expected outcome: Dockerfiles and Documentation
```
Comment (natalieparellano) [inline summerofcode/2023.md:60]:
```suggestion
- Expected outcome: updating the `pack` implementation to be more performant by taking advantage of the Docker daemon and updating documentation and sample workflows to reflect the changes
```

File: summerofcode/2023.md [hunk 5]
```diff
@@ -26,12 +26,42 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Mentor(s): Jesse Brown (@jabrown85); Juan Bustamante (@jjbustamante)
```
Comment (natalieparellano) [inline summerofcode/2023.md:61]:
```suggestion
- Mentor(s): Jesse Brown (@jabrown85); Natalie Arellano (@natalieparellano)
```

File: summerofcode/2023.md [hunk 6]
```diff
@@ -26,12 +26,42 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Difficulty: Medium
+- Upstream Issue (URL): why did we we let the new guy write this?
```
Comment (natalieparellano) [inline summerofcode/2023.md:63]:
```suggestion
- Upstream Issue (URL): https://github.com/buildpacks/pack/issues/1623
```

File: summerofcode/2023.md [hunk 7]
```diff
@@ -25,13 +25,39 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+#### ARM Support
+- Description: Widely anticipated since [the 1450s](https://en.wikipedia.org/wiki/Missa_L%27homme_arm%C3%A9), the rise of ARM processors has created new binary targets for pre-compiled executables. Creating these executables can be a valuable educational lesson about where assumptions get baked in, and what's really the difference between x86 and ARM anyhow?
+- Expected outcome: Improved ARM support
+- Recommended Skills: Golang, software development literacy.
+- Mentor(s): Aidan Delaney (@AidanDelaney); Jerico Pena (@jpena-r7)
+- Expected project size: 350 Hours
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/buildpacks/pack/issues/1459
+
+#### Something about Dockerfiles
```
Comment (natalieparellano) [inline summerofcode/2023.md:53]:
```suggestion
#### Enhancements for Dockerfiles
```

File: summerofcode/2023.md [hunk 8]
```diff
@@ -25,13 +25,39 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Mentor(s): Natalie Arellano (@natalieparellano); Joe Kimmel (@joe-kimmel-vmw)
+- Expected project size: 350 Hours
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/buildpacks/lifecycle/issues/993
+
+#### ARM Support
+- Description: Widely anticipated since [the 1450s](https://en.wikipedia.org/wiki/Missa_L%27homme_arm%C3%A9), the rise of ARM processors has created new binary targets for pre-compiled executables. Creating these executables can be a valuable educational lesson about where assumptions get baked in, and what's really the difference between x86 and ARM anyhow?
+- Expected outcome: Improved ARM support
+- Recommended Skills: Golang, software development literacy.
+- Mentor(s): Aidan Delaney (@AidanDelaney); Jerico Pena (@jpena-r7)
```
Comment (natalieparellano) [inline summerofcode/2023.md:48]:
```suggestion
- Mentor(s): Aidan Delaney (@AidanDelaney); Jerico Pena (@jpena-r7); Jesse Brown (@jabrown85)
```

**Comment by nate-double-u** [comment]
I don’t want to rush things, there is still time, but we’re in a new phase and having your proposal in the main list will become more valuable as potential contributors are looking over and deciding which projects to work on.

Take a look at discussion https://github.com/cncf/mentoring/discussions/848 for details on the timeline.

File: summerofcode/2023.md [hunk 9]
```diff
@@ -25,13 +25,39 @@ If you are a project maintainer and consider mentoring during the GSoC 2023 cycl
...
+- Upstream Issue (URL): https://github.com/buildpacks/lifecycle/issues/993
+
+#### ARM Support
+- Description: Widely anticipated since [the 1450s](https://en.wikipedia.org/wiki/Missa_L%27homme_arm%C3%A9), the rise of ARM processors has created new binary targets for pre-compiled executables. Creating these executables can be a valuable educational lesson about where assumptions get baked in, and what's really the difference between x86 and ARM anyhow?
+- Expected outcome: Improved ARM support
+- Recommended Skills: Golang, software development literacy.
+- Mentor(s): Aidan Delaney (@AidanDelaney); Jerico Pena (@jpena-r7); Jesse Brown (@jabrown85)
+- Expected project size: 350 Hours
+- Difficulty: Medium
+- Upstream Issue (URL): https://github.com/buildpacks/pack/issues/1459
```
Comment (natalieparellano) [inline summerofcode/2023.md:51]:
```suggestion
```

**Comment by nate-double-u** [comment]
Added a ToC entry, and moved the entries to match alphabetical order.

**Review by nate-double-u — APPROVED** [review]
Thanks for this!

**Pull request merged**